### PR TITLE
DRILL-5941: Added tests to verify correctness of counting lines when …

### PR DESCRIPTION
**Pull request by agozhiy**
…using a hive table with skipping header/footer for large datasets

### CODE REVIEW

File: framework/resources/Functional/hive/hive_storage/skipHeaderFooter/hiveHeaderFooter.json [hunk 1]
```diff
@@ -18,10 +18,20 @@
         }
     ],
     "datasources": [
-    { 
-       "mode": "gen", 
-       "src": "Datasources/hive/execHive.sh framework/resources/Datasources/hive_storage/skipHeaderFooter/skipHeaderFooter.ddl", 
-       "dest": "" 
-    }
+      {
+        "mode": "cp",
```
Comment (Agirish) [inline framework/resources/Functional/hive/hive_storage/skipHeaderFooter/hiveHeaderFooter.json]: Hey @agozhiy , can you please explain why we are copying the data here (cp) as well as in the (gen) script?

**Comment by Agirish** [comment]
Btw, Run went through fine. Once review comments are addressed, i'll have this merged.

File: framework/resources/Functional/hive/hive_storage/skipHeaderFooter/hiveHeaderFooter.json (hunk 1 above)
Comment (agozhiy) [reply framework/resources/Functional/hive/hive_storage/skipHeaderFooter/hiveHeaderFooter.json]: You are right, using "cp" is redundant.

**Comment by agozhiy** [comment]
I removed unnecessary "cp" from the test definition json, and did small changes to the datagen script so it will only put the data set to the dfs if it is not already present there. 
Please review

File: framework/resources/Functional/hive/hive_storage/skipHeaderFooter/hiveHeaderFooter.json [hunk 2]
```diff
@@ -18,10 +18,15 @@
...
+    {
+       "mode": "gen",
+       "src": "Datasources/hive/execHive.sh framework/resources/Datasources/hive_storage/skipHeaderFooter/skipHeaderFooter.ddl",
+       "dest": ""
+    },
+      {
+        "mode": "gen",
+        "src": "Datasources/hive_storage/skipHeaderFooter/DRILL_5941_prepare_data.sh",
+        "dest": ""
+      }
```
Comment (Agirish) [inline framework/resources/Functional/hive/hive_storage/skipHeaderFooter/hiveHeaderFooter.json]: Can you please fix the formatting (the two gen sections are not aligned)

File: framework/resources/Datasources/hive_storage/skipHeaderFooter/DRILL_5941_prepare_data.sh [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        untar_data $location $tar_name $file_name
+    fi
+
+    copy_to_dfs $location $file_name $destination
+}
+
+tar_name="DRILL_5941_data.tar.gz"
+prepare_dataset $test_dir "DRILL_5941_lines_count_skip_header_footer_big_data_set.csv" $tar_name $dfs_location
+
+${DRILL_TEST_DATA_DIR}/Datasources/hive/execHive.sh framework/resources/Datasources/hive_storage/skipHeaderFooter/DRILL_5941_lines_count_skip_header_footer_create_tables.ddl
```
Comment (Agirish) [inline framework/resources/Datasources/hive_storage/skipHeaderFooter/DRILL_5941_prepare_data.sh]: Can you change framework/resources to $DRILL_TEST_DATA_DIR

File: framework/resources/Functional/hive/hive_storage/skipHeaderFooter/hiveHeaderFooter.json (hunk 2 above)
Comment (agozhiy) [reply framework/resources/Functional/hive/hive_storage/skipHeaderFooter/hiveHeaderFooter.json]: Done.

File: framework/resources/Datasources/hive_storage/skipHeaderFooter/DRILL_5941_prepare_data.sh (hunk 3 above)
Comment (agozhiy) [reply framework/resources/Datasources/hive_storage/skipHeaderFooter/DRILL_5941_prepare_data.sh]: Done.

**Comment by agozhiy** [comment]
Made fixes according to your comments. Please review

**Comment by Agirish** [comment]
Thanks Anton. Scheduling another run to make sure there are no runtime issues.

**Comment by Agirish** [comment]
Run went through fine. Merging.

**Pull request merged**